### PR TITLE
ssh/tailssh: add support for sftp

### PR DESCRIPTION
**Pull request by maisem**
Updates #3802

Signed-off-by: Maisem Ali <maisem@tailscale.com>

### CODE REVIEW

File: ssh/tailssh/tailssh.go [hunk 1]
```diff
@@ -813,27 +815,29 @@ func (ss *sshSession) run() {
...
 	var rec *recording // or nil if disabled
-	if ss.shouldRecord() {
-		var err error
-		rec, err = ss.startNewRecording()
-		if err != nil {
-			fmt.Fprintf(ss, "can't start new recording\r\n")
-			ss.logf("startNewRecording: %v", err)
-			ss.Exit(1)
-			return
+	if ss.Subsystem() != "sftp" {
```
Comment (bradfitz) [inline ssh/tailssh/tailssh.go:819]: switch on this with a default to panic or bail out if not sftp/empty string?

**Review by bradfitz — APPROVED** [review]
Nice!

Which clients have you tested it with?

File: ssh/tailssh/incubator.go [hunk 2]
```diff
@@ -76,20 +90,38 @@ func (ss *sshSession) newIncubatorCommand(ctx context.Context, name string, args
...
 		"--remote-user=" + remoteUser,
 		"--remote-ip=" + ci.src.IP().String(),
-		"--cmd=" + name,
 		"--has-tty=false", // updated in-place by startWithPTY
 		"--tty-name=",     // updated in-place by startWithPTY
 	}
-	if len(args) > 0 {
-		incubatorArgs = append(incubatorArgs, "--")
-		incubatorArgs = append(incubatorArgs, args...)
+	if ss.Subsystem() == "sftp" {
```
[the remark this answers is not in the document]
Comment (maisem) [reply ssh/tailssh/incubator.go]: Added a panic

File: ssh/tailssh/tailssh.go (hunk 1 above)
Comment (maisem) [reply ssh/tailssh/tailssh.go:828]: Done

File: ssh/tailssh/tailssh.go [hunk 3]
```diff
@@ -221,10 +221,12 @@ func (c *conn) ServerConfig(ctx ssh.Context) *gossh.ServerConfig {
...
 	c.Server = &ssh.Server{
-		Version:           "Tailscale",
-		Handler:           c.handleConnPostSSHAuth,
-		RequestHandlers:   map[string]ssh.RequestHandler{},
-		SubsystemHandlers: map[string]ssh.SubsystemHandler{},
+		Version:         "Tailscale",
+		Handler:         c.handleConnPostSSHAuth,
+		RequestHandlers: map[string]ssh.RequestHandler{},
+		SubsystemHandlers: map[string]ssh.SubsystemHandler{
+			"sftp": c.handleConnPostSSHAuth,
```
[the remark this answers is not in the document]
Comment (maisem) [reply ssh/tailssh/tailssh.go:228]: Done

**Pull request merged**